salt/cloud/clouds/ec2.py

```diff
@@ -2195,22 +2196,17 @@ def query_instance(vm_=None, call=None):
                 'There was an error in the query. %s attempts '
                 'remaining: %s', attempts, data['error']
             )
-            attempts -= 1
-            # Just a little delay between attempts...
-            time.sleep(1)
-            continue
-
-        if isinstance(data, list) and not data:
+        elif isinstance(data, list) and not data:
             log.warning(
                 'Query returned an empty list. %s attempts '
                 'remaining.', attempts
             )
-            attempts -= 1
-            # Just a little delay between attempts...
-            time.sleep(1)
-            continue
+        else:
+            break
 
-        break
+        time.sleep(2**attempts)
```

Comment:
Can you switch this to `time.sleep(random.uniform(1, 2**attempts))` like is done in the other pr?

Thanks,
Daniel

Reply:
Thanks for the hint. I updated the PR.